### DISCUSSION TITLE
Bluetooth Controller with Extended Advertising

### DISCUSSION THREAD

**Discussion by LingaoM** [zephyrproject-rtos/zephyr#49860] [Q&A]
Hi,

i'm use Zephyr Bluetooth Controller to develop bluetooth extended advertising application, but found the secondary advertising pdu aux offset too big than Nordic Connect SDK.

For example, ncs sdk aux offset almost 400us, but the current zephyr sdk bigger than 10ms

![image](https://user-images.githubusercontent.com/26378606/188043193-8d7e91de-d9ed-4994-9a21-56a9d56a8b67.png)

**Comment by LingaoM** [comment]
[ncs_ext_adv_demo.zip](https://github.com/zephyrproject-rtos/zephyr/files/9476330/ncs_ext_adv_demo.zip)

**Comment by cvinayak** [comment]
In the Zephyr Controller design, the scheduling of AUX_ADV_IND PDUs are different to ensure that they can be placed non-overlapping with other states and roles, and the ADV_EXT_IND PDUs would be scheduled with the required random_delay.

**Reply by LingaoM** [reply]
Hi，thanks your help.

I'm want to change zephyr Bluetooth controller so that aux offset as minimum as possible, can you tell me i need to change where?

I'm focus on based Extended Advertising Mesh networking, so currently aux offset to big that cause mesh networks delay.

**Reply by cvinayak** [reply]
The aux offset range from ~600 us upto advertising interval plus 10ms random delay. First advertising interval would have aux offset ~600 us (this can be tuned though but not below 300us). But consecutive event's AUX_ADV_IND would be at advertising interval plus random delay of 10ms.

See the below, the AUX_ADV_IND are every 30 ms for a 20 ms advertising interval.

![image](https://user-images.githubusercontent.com/6350656/188294076-6668d2bc-c769-4520-bccc-4f26e37217ef.png)

I do not understand how there would be mesh network delay as long as the advertising events are all following the BT Spec defined advertising interval plus 10 ms advertising delay. Yes, when random delay of lower than 10ms then Zephyr Controller would have lower delay, but worst case delays would be a deterministic 10 ms per advertising interval.

**Reply by LingaoM** [reply]
This type is more like ADV_PERIODIC_IND.

**Reply by LingaoM** [reply]
For Mesh Relay messages, time conflicts will inevitably occur, resulting in packet loss.

**Reply by LingaoM** [reply]
Imagine that multiple Mesh Relay nodes receive a message and Relay at the same time, even if ADV_EXT_IND is random, but in Zephyr's implementation, ADV_AUX_IND is timed, so multiple Relay messages cannot be received by the same device, because they Time all collides.

**Reply by cvinayak** [reply]
I agree that when multiple nodes advertise at the same moment, the AUX_ADV_IND would overlap but each one will use random radio channel index (uses CSA#2). Devices listening would receive ADV_EXT_IND at random time and listen to AUX_ADV_IND at the specific radio channel idx.

**Reply by LingaoM** [reply]
Although channel differences, but the time is same, so only one aux advertisement can received, other message will be discarded unexpectedly.

**Reply by cvinayak** [reply]
It is the same when all the ADV_EXT_IND overlap when multiple nodes advertising at the same time. Only one primary channel PDU is received hence one AUX_ADV_IND when there are overlap.

I do not disagree that current Zephyr Controller takes worst case delay in receiving all the advertising from multiple nodes.

Zephyr Controller's is focused towards maximizing the radio utilization for audio uses cases where each radio use be smallest to increase available radio for ISO radio events.  No optimization (controller) for Mesh stack is planned yet.
